[docs] Alternatives for Expo Notifications (#34705)
expo/expo: docs/pages/guides/notifications-services.mdx

---

> title: Notifications

Reviewer: ```suggestion
title: Using Push Notifications
```

Let's change this title to follow the convention in the "using" integration section: 

![CleanShot 2025-02-11 at 00 56 32@2x](https://github.com/user-attachments/assets/9be588d0-7d8a-4bd8-93fc-2e0a9dfc4dc2)

description: Learn about different push notification services.
---

import { BoxLink } from '~/ui/components/BoxLink';

Expo apps can work with any notification service or capabilities offered by the operating system thanks to [config plugins](/config-plugins/introduction/). The following services provide purpose-built Expo integrations for implementing push notifications in your app:

> The [`expo-notifications`](/versions/latest/sdk/notifications/) library is designed and tested to work with Expo's push notification service. Some advanced features may not be compatible with third-party providers, as they often have their own native and React Native SDKs optimized for their services.

> ### Expo Notifications

Reviewer: ```suggestion
## Expo push notifications
```

Let's update this title to reflect the service name instead of the library.

> ### Expo Notifications

Reviewer: Let's update these H3s to H2s and H4s to H3s in this guide. Generally, having semantic components following their order helps with SEO a lot.

> [Expo Notifications](/push-notifications/overview) provides a unified API for handling push notifications across Android and iOS. It integrates seamlessly with your Expo account and is free to use.

Reviewer: ```suggestion
[Expo Notifications](/versions/latest/sdk/notifications/) provides a unified API for handling push notifications across Android and iOS. It integrates seamlessly with your Expo account and is free to use.
```

The link goes to the Push notifications service page instead of the library where as in the content we referencing the library.

Reviewer: Maybe you want to change it to Expo push notifications?

#### Key capabilities:

- Fully compatible with the [`expo-notifications`](/versions/latest/sdk/notifications/) library
- Includes an Expo Dashboard to track notifications and user engagement
- Supports testing notifications with the [Expo Notifications Tool](https://expo.dev/notifications)

#### Considerations and limitations:

- Does not support advanced platform features like notification extensions
- Limited to 600 notifications per second per project, which works for most apps but may not suit high-volume scenarios

> For implementation details, see the [Expo push notifications overview](/push-notifications/overview) and [server-side SDK options](https://docs.expo.dev/push-notifications/sending-notifications/#send-push-notifications-using-a-servers).

Reviewer: ```suggestion
For implementation details, see the [Expo push notifications overview](/push-notifications/overview) and [server-side SDK options](/push-notifications/sending-notifications/#send-push-notifications-using-a-servers).
```

- Internal link
- Also let's convert this to two BoxLinks.

### OneSignal

> [OneSignal](https://onesignal.com/) is a customer engagement platform that provides push notifications, in-app messaging, SMS, and email services for web and mobile apps. It includes an [Expo SDK](https://github.com/OneSignal/onesignal-expo-plugin) for easy integration. Follow the guide below for a step-by-step setup.

Reviewer: ```suggestion
[OneSignal](https://onesignal.com/) is a customer engagement platform that provides push notifications, in-app messaging, SMS, and email services for web and mobile apps. It includes an [Expo SDK](https://github.com/OneSignal/onesignal-expo-plugin) for easy integration.
```

We have already specified the BoxLink so let's remove this line.

#### Key capabilities:

- Supports [Notification Service Extensions](https://documentation.onesignal.com/docs/service-extensions) for Android and iOS
- Provides a custom dashboard for managing notifications and user engagement

#### Considerations and limitations:

- Not free. OneSignal offers free limits, but advanced features (e.g., segmentation, automation) require a paid plan. See the [OneSignal pricing page](https://onesignal.com/pricing).
- Depends on OneSignal to maintain their Expo plugin and SDK.

> <BoxLink
>   title="OneSignal Expo SDK Setup"
>   description="Learn how to integrate OneSignal in your Expo project."
>   href="https://documentation.onesignal.com/docs/react-native-expo-sdk-setup"
> />

Reviewer: ```suggestion
<BoxLink
  title="OneSignal Expo SDK Setup"
  description="Follow this guide for a step-by-step setup on how to integrate OneSignal in your Expo project."
  href="https://documentation.onesignal.com/docs/react-native-expo-sdk-setup"
/>
```

Let's make the call to action a bit more clear here.

### Braze

[Braze](https://www.braze.com/) is a customer engagement platform that delivers personalized, cross-channel messaging through push notifications, in-app messaging, email, SMS, and web. It provides a [React Native SDK](https://github.com/braze-inc/braze-react-native-sdk) and an [Expo plugin](https://github.com/braze-inc/braze-expo-plugin/tree/main). Check out the [Expo example app](https://github.com/braze-inc/braze-expo-plugin/tree/main/example) for more details.

#### Key capabilities:

- Supports customizable push notifications with images, videos, and location
- Includes a dashboard for managing campaigns and journey orchestration

#### Considerations and limitations:

- Requires contacting Sales for pricing information
- Depends on Braze to maintain their Expo plugin and SDK

> <BoxLink
>   title="Braze Expo Setup"
>   description="Learn how to integrate Braze in your Expo project."
>   href="https://www.braze.com/docs/developer_guide/platforms/react_native/sdk_integration"
> />

Reviewer: ```suggestion
<BoxLink
  title="Braze Expo Setup"
  description="Follow this guide for a step-by-step setup on how to integrate Braze in your Expo project."
  href="https://www.braze.com/docs/developer_guide/platforms/react_native/sdk_integration"
/>
```

### Customer.io

> [Customer.io](http://Customer.io) provides two main services: **Journeys**, for setting up campaigns and sending messages, emails, and push notifications, and **Data Pipelines**, for capturing and forwarding data from your Expo app, website, or other sources to the right destinations in your stack. It offers an [Expo plugin](https://github.com/customerio/customerio-expo-plugin) for seamless integration.

Reviewer: ```suggestion
[Customer.io](http://Customer.io) provides two main services: 
- **Journeys**, for setting up campaigns and sending messages, emails, and push notifications
- **Data Pipelines**, for capturing and forwarding data from your Expo app, website, or other sources to the right destinations in your stack

It offers an [Expo plugin](https://github.com/customerio/customerio-expo-plugin) for seamless integration.
```

both features have multiple capabilities so let's cover it into list like this for easier readability

#### Key capabilities:

- Supports customizable push notifications with images, videos, and location
- Includes a dashboard for managing campaigns and journey orchestration

#### Considerations and limitations:

- Not free. Customer.io does not offer a free tier. The "Essentials" plan starts at $100/month. See the [Customer.io pricing page](https://customer.io/pricing) for more details.
- Depends on Customer.io to maintain their Expo plugin and SDK.

> <BoxLink
>   title="Customer.io Expo Quick Start Guide"
>   description="Learn how to integrate Customer.io in your Expo project."
>   href="https://docs.customer.io/sdk/expo/quick-start-guide/"
> />

Reviewer: ```suggestion
<BoxLink
  title="Customer.io Expo Quick Start Guide"
  description="Follow this guide for a step-by-step setup on how to integrate Customer.io in your Expo project."
  href="https://docs.customer.io/sdk/expo/quick-start-guide/"
/>
```

---
title: Using Push Notifications
description: Learn about different push notification services.
---

import { BoxLink } from '~/ui/components/BoxLink';

> Expo apps can work with any notification service or capabilities offered by the operating system thanks to [config plugins](/config-plugins/introduction/). The following services provide purpose-built Expo integrations for implementing push notifications in your app:

Reviewer 2: ```suggestion
Expo apps can work with any notification service or capabilities offered by the operating system. The following services provide purpose-built Expo integrations for implementing push notifications in your app:
```

the role of config plugins is not directly visible to me here. Either remove or expand on the relation.

> The [`expo-notifications`](/versions/latest/sdk/notifications/) library is designed and tested to work with Expo's push notification service. Some advanced features may not be compatible with third-party providers, as they often have their own native and React Native SDKs optimized for their services.

## Expo push notifications

[Expo Notifications](/versions/latest/sdk/notifications/) provides a unified API for handling push notifications across Android and iOS. It integrates seamlessly with your Expo account and is free to use.

#### Key capabilities:

- Fully compatible with the [`expo-notifications`](/versions/latest/sdk/notifications/) library
- Includes an Expo Dashboard to track notifications and user engagement
- Supports testing notifications with the [Expo Notifications Tool](https://expo.dev/notifications)

#### Considerations and limitations:

> - Does not support advanced platform features like notification extensions

Reviewer 2: re notification extensions:

I believe that if people want to have a notification extension, they can have it. To my knowledge, we're not preventing them from doing that. Listing this as a limitation is inaccurate. I can test it later to be sure but I tend to think this should be removed.

Reviewer 3: I've definitely heard of people doing it, and it was more like "we can't guarantee this works". If we test it and it works, that would be great to note - not that we're suddenly supporting it, but we could adjust this message to something like "the library doesn't have anything built-in for this, it's not supported, but nothing is stopping you"

Reviewer 2: Notification extensions are integrated pretty deeply into iOS, and I think we can guarantee they work to the same extent as Apple :D. But I will double-check next week. I think we can say that "iOS notification extension is supported but you need to set it up yourself /according to this guide/. (we don't have a guide for that). But saying they are not supported is not the right message.
